File: web/tests/reference/binary-trees/diameter-of-binary-tree.js
function diameterOfBinaryTree(root) {
  let best = 0;
  const height = (node) => {
    if (node === null) return 0;
    const left = height(node.left);
    const right = height(node.right);
    if (left + right > best) best = left + right;
    return 1 + Math.max(left, right);
  };
  height(root);
  return best;
}

File: web/tests/reference/binary-trees/diameter-of-binary-tree.py
def diameter_of_binary_tree(root):
    best = [0]

    def height(node):
        if node is None:
            return 0
        left = height(node.left)
        right = height(node.right)
        if left + right > best[0]:
            best[0] = left + right
        return 1 + max(left, right)

    height(root)
    return best[0]
